store.go

```diff
@@ -1783,13 +1759,13 @@ func (s *store) CreateContainer(id string, names []string, image, layer, metadat
 	var options ContainerOptions
 	if cOptions != nil {
 		options = *cOptions
-		options.IDMappingOptions.UIDMap = copyIDMap(cOptions.IDMappingOptions.UIDMap)
-		options.IDMappingOptions.GIDMap = copyIDMap(cOptions.IDMappingOptions.GIDMap)
-		options.LabelOpts = copyStringSlice(cOptions.LabelOpts)
+		options.IDMappingOptions.UIDMap = copySlicePreferringNil(cOptions.IDMappingOptions.UIDMap)
+		options.IDMappingOptions.GIDMap = copySlicePreferringNil(cOptions.IDMappingOptions.GIDMap)
+		options.LabelOpts = copySlicePreferringNil(cOptions.LabelOpts)
 		options.Flags = maps.Clone(cOptions.Flags)
-		options.MountOpts = copyStringSlice(cOptions.MountOpts)
+		options.MountOpts = copySlicePreferringNil(cOptions.MountOpts)
 		options.StorageOpt = copyStringStringMap(cOptions.StorageOpt)
-		options.BigData = copyContainerBigDataOptionSlice(cOptions.BigData)
+		options.BigData = slices.Clone(cOptions.BigData)
```

Comment:
`slices.Clone` makes a shallow clone, i.e. `Data` slice is not really copied with this change.

Reply:
The previous `copyContainerBigDataOptionSlice` function used the `slices.Clone` function to copy data. So I would expect similar behavior.

Reply:
It used `slices.Clone()` to copy the `Data` field in each element in the slice.  The godoc for `slices.Clone()` notes that it only copies using assignment, so a new slice would not be created for `Data` fields.